build.sh

```diff
@@ -36,8 +36,10 @@ as_root ldconfig
 
 # Run unit tests
 export LD_LIBRARY_PATH="${PREFIX_DIR}/usr/local/lib:/usr/local/lib64"
-ctest --output-on-failure --test-dir "${MCM_BUILD_DIR}" -V
-
+"${MCM_BUILD_DIR}/bin/sdk_unit_tests"
+"${MCM_BUILD_DIR}/bin/media_proxy_unit_tests"
```

Comment:
Is this change in build.sh expected ?

Reply:
Yep. For me `ctest` couldn't find any test binaries for some reason.

Reply:
Ok, then please add those unit_test binaries too:
conn_rdma_base_unit_tests
conn_rdma_real_ep_test
conn_rdma_rx_tx_unit_tests

Reply:
I can't because they fail actually
```
[==========] Running 18 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 18 tests from RdmaTest
[ RUN      ] RdmaTest.ConfigureSuccess
[       OK ] RdmaTest.ConfigureSuccess (0 ms)
[ RUN      ] RdmaTest.EstablishSuccess
[       OK ] RdmaTest.EstablishSuccess (0 ms)
[ RUN      ] RdmaTest.EstablishFailureEpInit
Dec 19 09:00:59.584 [ERRO] Failed to initialize RDMA endpoint context error="Operation not permitted"
[       OK ] RdmaTest.EstablishFailureEpInit (0 ms)
[ RUN      ] RdmaTest.CleanupResources
[       OK ] RdmaTest.CleanupResources (0 ms)
[ RUN      ] RdmaTest.ValidateStateTransitions
[       OK ] RdmaTest.ValidateStateTransitions (0 ms)
[ RUN      ] RdmaTest.InitQueueWithElementsSuccess
[       OK ] RdmaTest.InitQueueWithElementsSuccess (0 ms)
[ RUN      ] RdmaTest.InitQueueWithElementsFailureMemoryAllocationWithSize0
Dec 19 09:00:59.585 [ERRO] Invalid transfer size provided for RDMA buffer allocation trx_sz=0
[       OK ] RdmaTest.InitQueueWithElementsFailureMemoryAllocationWithSize0 (0 ms)
[ RUN      ] RdmaTest.InitQueueWithElementsFailureMemoryAllocationWithSize1GB
Dec 19 09:00:59.585 [ERRO] Invalid transfer size provided for RDMA buffer allocation trx_sz=18446744071562067968
[       OK ] RdmaTest.InitQueueWithElementsFailureMemoryAllocationWithSize1GB (0 ms)
[ RUN      ] RdmaTest.AddToQueueSuccess
[       OK ] RdmaTest.AddToQueueSuccess (0 ms)
[ RUN      ] RdmaTest.AddToQueueFailureNullptr
[       OK ] RdmaTest.AddToQueueFailureNullptr (0 ms)
[ RUN      ] RdmaTest.ConsumeFromQueueSuccess
[       OK ] RdmaTest.ConsumeFromQueueSuccess (0 ms)
[ RUN      ] RdmaTest.ConsumeFromQueueFailureEmptyQueue
[       OK ] RdmaTest.ConsumeFromQueueFailureEmptyQueue (0 ms)
[ RUN      ] RdmaTest.ConsumeFromQueueFailureContextCancelled
[       OK ] RdmaTest.ConsumeFromQueueFailureContextCancelled (0 ms)
[ RUN      ] RdmaTest.CleanupQueue
[       OK ] RdmaTest.CleanupQueue (0 ms)
[ RUN      ] RdmaTest.QueuePerformanceTest
Processed 1000000 queue operations in 0.0197483 seconds.
[       OK ] RdmaTest.QueuePerformanceTest (1649 ms)
[ RUN      ] RdmaTest.RepeatedShutdown
Dec 19 09:01:01.237 [INFO] RDMA device successfully deinitialized
[       OK ] RdmaTest.RepeatedShutdown (0 ms)
[ RUN      ] RdmaTest.StressTest
[       OK ] RdmaTest.StressTest (3170 ms)
[ RUN      ] RdmaTest.ConcurrentAccessWithDelays
[       OK ] RdmaTest.ConcurrentAccessWithDelays (379 ms)
[----------] 18 tests from RdmaTest (5204 ms total)

[----------] Global test environment tear-down
[==========] 18 tests from 1 test suite ran. (5204 ms total)
[  PASSED  ] 18 tests.
Running main() from /media/disk/kilichev/projects/03-Audio/Media-Communications-Mesh/_build/mcm/_deps/googletest-src/googletest/src/gtest_main.cc
[==========] Running 2 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 2 tests from RdmaRealEndpointsTest
[ RUN      ] RdmaRealEndpointsTest.ConcurrentTransmissionAndReception
Dec 19 09:01:04.900 [DEBU] RdmaRx configure local_ip="192.168.1.22" local_port="9002" remote_ip="" remote_port=""
Interface for IP 192.168.1.22 not found
fi_getinfo(): /media/disk/kilichev/projects/03-Audio/Media-Communications-Mesh/media-proxy/src/libfabric_ep.c:317, ret=-61 (No data available)
Dec 19 09:01:04.941 [ERRO] Failed to initialize RDMA endpoint context error="No data available"
/media/disk/kilichev/projects/03-Audio/Media-Communications-Mesh/media-proxy/tests/conn_rdma_real_ep_test.cc:148: Failure
Expected equality of these values:
  conn_rx->establish(ctx)
    Which is: 4-byte object <09-00 00-00>
  connection::Result::success
    Which is: 4-byte object <00-00 00-00>
```
It'll be a task for Maciej in the next cycle.

Reply:
conn_rdma_real_ep_test is just for real device testing, it was never added to the tests that were running after each build:

# ONLY USE THIS TEST (SET ON OPTION) IF YOU HAVE CONFIGURED REAL RDMA ENDPOINTS
# IN FUNCTION SetupRdmaConnections
option(RUN_REAL_RDMA_PERFORMANCE_TESTS "Build and enable unit tests" OFF)
if (RUN_REAL_RDMA_PERFORMANCE_TESTS)
    add_test(NAME conn_rdma_real_ep_test COMMAND conn_rdma_real_ep_test)
    set_tests_properties(conn_rdma_real_ep_test PROPERTIES DESCRIPTION "
        Validates the functionality, performance, and correctness of RDMA Tx and Rx operations with 
            real endpoint configurations:
        1. Tests concurrent data transmission and reception between an emulated transmitter
            and a test receiver.
        2. Measures latency and bandwidth for varying payload sizes and queue depths.
        3. Reports average latency and transfer rate in GB/s for different configurations,
            ensuring RDMA endpoint reliability and efficiency.
    ")

Reply:
Oh, then we need to add only 2 binaries:
conn_rdma_base_unit_tests
conn_rdma_rx_tx_unit_tests

Reply:
They need to be fixed first.
```
[==========] Running 18 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 18 tests from RdmaTest
[ RUN      ] RdmaTest.ConfigureSuccess
[       OK ] RdmaTest.ConfigureSuccess (0 ms)
[ RUN      ] RdmaTest.EstablishSuccess
[       OK ] RdmaTest.EstablishSuccess (0 ms)
[ RUN      ] RdmaTest.EstablishFailureEpInit
Dec 19 11:46:13.166 [ERRO] Failed to initialize RDMA endpoint context error="Operation not permitted"
[       OK ] RdmaTest.EstablishFailureEpInit (0 ms)
[ RUN      ] RdmaTest.CleanupResources
[       OK ] RdmaTest.CleanupResources (0 ms)
[ RUN      ] RdmaTest.ValidateStateTransitions
[       OK ] RdmaTest.ValidateStateTransitions (0 ms)
[ RUN      ] RdmaTest.InitQueueWithElementsSuccess
[       OK ] RdmaTest.InitQueueWithElementsSuccess (0 ms)
[ RUN      ] RdmaTest.InitQueueWithElementsFailureMemoryAllocationWithSize0
Dec 19 11:46:13.168 [ERRO] Invalid transfer size provided for RDMA buffer allocation trx_sz=0
[       OK ] RdmaTest.InitQueueWithElementsFailureMemoryAllocationWithSize0 (0 ms)
[ RUN      ] RdmaTest.InitQueueWithElementsFailureMemoryAllocationWithSize1GB
Dec 19 11:46:13.168 [ERRO] Invalid transfer size provided for RDMA buffer allocation trx_sz=18446744071562067968
[       OK ] RdmaTest.InitQueueWithElementsFailureMemoryAllocationWithSize1GB (0 ms)
[ RUN      ] RdmaTest.AddToQueueSuccess
[       OK ] RdmaTest.AddToQueueSuccess (0 ms)
[ RUN      ] RdmaTest.AddToQueueFailureNullptr
[       OK ] RdmaTest.AddToQueueFailureNullptr (0 ms)
[ RUN      ] RdmaTest.ConsumeFromQueueSuccess
[       OK ] RdmaTest.ConsumeFromQueueSuccess (0 ms)
[ RUN      ] RdmaTest.ConsumeFromQueueFailureEmptyQueue
[       OK ] RdmaTest.ConsumeFromQueueFailureEmptyQueue (0 ms)
[ RUN      ] RdmaTest.ConsumeFromQueueFailureContextCancelled
[       OK ] RdmaTest.ConsumeFromQueueFailureContextCancelled (0 ms)
[ RUN      ] RdmaTest.CleanupQueue
[       OK ] RdmaTest.CleanupQueue (0 ms)
[ RUN      ] RdmaTest.QueuePerformanceTest
Processed 1000000 queue operations in 0.0194703 seconds.
[       OK ] RdmaTest.QueuePerformanceTest (2487 ms)
[ RUN      ] RdmaTest.RepeatedShutdown
Dec 19 11:46:15.657 [INFO] RDMA device successfully deinitialized
[       OK ] RdmaTest.RepeatedShutdown (0 ms)
[ RUN      ] RdmaTest.StressTest
[       OK ] RdmaTest.StressTest (3174 ms)
[ RUN      ] RdmaTest.ConcurrentAccessWithDelays
[       OK ] RdmaTest.ConcurrentAccessWithDelays (368 ms)
[----------] 18 tests from RdmaTest (6034 ms total)

[----------] Global test environment tear-down
[==========] 18 tests from 1 test suite ran. (6034 ms total)
[  PASSED  ] 18 tests.
Running main() from gmock_main.cc
[==========] Running 10 tests from 2 test suites.
[----------] Global test environment set-up.
[----------] 5 tests from RdmaRxTest
[ RUN      ] RdmaRxTest.EstablishSuccess
Dec 19 11:46:19.313 [DEBU] RdmaRx configure local_ip="192.168.1.10" local_port="8001" remote_ip="192.168.1.20" remote_port="8002"
Dec 19 11:46:19.313 [INFO] RDMA device successfully deinitialized
[       OK ] RdmaRxTest.EstablishSuccess (0 ms)
[ RUN      ] RdmaRxTest.EstablishFailureEpInit
Dec 19 11:46:19.313 [DEBU] RdmaRx configure local_ip="192.168.1.10" local_port="8001" remote_ip="192.168.1.20" remote_port="8002"
Dec 19 11:46:19.313 [ERRO] Failed to initialize RDMA endpoint context error="Operation not permitted"
[       OK ] RdmaRxTest.EstablishFailureEpInit (0 ms)
[ RUN      ] RdmaRxTest.EstablishFailureBufferAllocation
Dec 19 11:46:19.314 [DEBU] RdmaRx configure local_ip="192.168.1.10" local_port="8001" remote_ip="192.168.1.20" remote_port="8002"
Dec 19 11:46:19.314 [ERRO] Memory registration failed for the RDMA buffer block error="Operation not permitted"
Dec 19 11:46:19.314 [ERRO] RDMA configuring failed state="closed"
[       OK ] RdmaRxTest.EstablishFailureBufferAllocation (0 ms)
[ RUN      ] RdmaRxTest.EstablishAlreadyInitialized
Dec 19 11:46:19.314 [DEBU] RdmaRx configure local_ip="192.168.1.10" local_port="8001" remote_ip="192.168.1.20" remote_port="8002"

GMOCK WARNING:
Uninteresting mock function call - returning default value.
    Function call: start_threads(@0x5629b645fc80 64-byte object <78-4A 4F-B4 29-56 00-00 20-D0 45-B6 29-56 00-00 A0-01 54-B4 29-56 00-00 A0-D3 45-B6 29-56 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 F0-07 46-B6 29-56 00-00>)
          Returns: 4-byte object <00-00 00-00>
NOTE: You can safely ignore the above warning unless this call should not happen.  Do not suppress it by blindly adding an EXPECT_CALL() if you don't mean to enforce the call.  See https://github.com/google/googletest/blob/main/docs/gmock_cook_book.md#knowing-when-to-expect-useoncall for details.

GMOCK WARNING:
Uninteresting mock function call - returning default value.
    Function call: rdma_deinit(0x7ffde0ddaf88)
          Returns: 0
NOTE: You can safely ignore the above warning unless this call should not happen.  Do not suppress it by blindly adding an EXPECT_CALL() if you don't mean to enforce the call.  See https://github.com/google/googletest/blob/main/docs/gmock_cook_book.md#knowing-when-to-expect-useoncall for details.
Dec 19 11:46:19.314 [INFO] RDMA device successfully deinitialized
[       OK ] RdmaRxTest.EstablishAlreadyInitialized (0 ms)
[ RUN      ] RdmaRxTest.ValidateStateTransitions
Dec 19 11:46:19.314 [DEBU] RdmaRx configure local_ip="192.168.1.10" local_port="8001" remote_ip="192.168.1.20" remote_port="8002"

GMOCK WARNING:
Uninteresting mock function call - returning default value.
    Function call: start_threads(@0x5629b645fc80 64-byte object <78-4A 4F-B4 29-56 00-00 90-1E 46-B6 29-56 00-00 A0-01 54-B4 29-56 00-00 E0-B8 43-B6 29-56 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 50-1F 46-B6 29-56 00-00>)
          Returns: 4-byte object <00-00 00-00>
NOTE: You can safely ignore the above warning unless this call should not happen.  Do not suppress it by blindly adding an EXPECT_CALL() if you don't mean to enforce the call.  See https://github.com/google/googletest/blob/main/docs/gmock_cook_book.md#knowing-when-to-expect-useoncall for details.
[       OK ] RdmaRxTest.ValidateStateTransitions (0 ms)
[----------] 5 tests from RdmaRxTest (1 ms total)

[----------] 5 tests from RdmaTxTest
[ RUN      ] RdmaTxTest.EstablishSuccess
Dec 19 11:46:19.314 [DEBU] RdmaTx configure local_ip="192.168.1.10" local_port="8001" remote_ip="192.168.1.20" remote_port="8002"

GMOCK WARNING:
Uninteresting mock function call - returning default value.
    Function call: start_threads(@0x5629b645fc80 64-byte object <78-4A 4F-B4 29-56 00-00 50-C9 45-B6 29-56 00-00 A0-01 54-B4 29-56 00-00 A0-D3 45-B6 29-56 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 F0-07 46-B6 29-56 00-00>)
          Returns: 4-byte object <00-00 00-00>
NOTE: You can safely ignore the above warning unless this call should not happen.  Do not suppress it by blindly adding an EXPECT_CALL() if you don't mean to enforce the call.  See https://github.com/google/googletest/blob/main/docs/gmock_cook_book.md#knowing-when-to-expect-useoncall for details.
Dec 19 11:46:19.314 [INFO] RDMA device successfully deinitialized
[       OK ] RdmaTxTest.EstablishSuccess (0 ms)
[ RUN      ] RdmaTxTest.EstablishFailureEpInit
Dec 19 11:46:19.315 [DEBU] RdmaTx configure local_ip="192.168.1.10" local_port="8001" remote_ip="192.168.1.20" remote_port="8002"
Dec 19 11:46:19.315 [ERRO] Failed to initialize RDMA endpoint context error="Operation not permitted"
[       OK ] RdmaTxTest.EstablishFailureEpInit (0 ms)
[ RUN      ] RdmaTxTest.EstablishFailureBufferAllocation
Dec 19 11:46:19.315 [DEBU] RdmaTx configure local_ip="192.168.1.10" local_port="8001" remote_ip="192.168.1.20" remote_port="8002"
Dec 19 11:46:19.315 [ERRO] Memory registration failed for the RDMA buffer block error="Operation not permitted"
Dec 19 11:46:19.315 [ERRO] RDMA configuring failed state="closed"
[       OK ] RdmaTxTest.EstablishFailureBufferAllocation (0 ms)
[ RUN      ] RdmaTxTest.EstablishAlreadyInitialized
Dec 19 11:46:19.315 [DEBU] RdmaTx configure local_ip="192.168.1.10" local_port="8001" remote_ip="192.168.1.20" remote_port="8002"

GMOCK WARNING:
Uninteresting mock function call - returning default value.
    Function call: start_threads(@0x5629b645fc80 64-byte object <78-4A 4F-B4 29-56 00-00 70-02 46-B6 29-56 00-00 A0-01 54-B4 29-56 00-00 E0-B8 43-B6 29-56 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 D0-17 46-B6 29-56 00-00>)
          Returns: 4-byte object <00-00 00-00>
NOTE: You can safely ignore the above warning unless this call should not happen.  Do not suppress it by blindly adding an EXPECT_CALL() if you don't mean to enforce the call.  See https://github.com/google/googletest/blob/main/docs/gmock_cook_book.md#knowing-when-to-expect-useoncall for details.

GMOCK WARNING:
Uninteresting mock function call - returning default value.
    Function call: rdma_deinit(0x7ffde0ddaf88)
          Returns: 0
NOTE: You can safely ignore the above warning unless this call should not happen.  Do not suppress it by blindly adding an EXPECT_CALL() if you don't mean to enforce the call.  See https://github.com/google/googletest/blob/main/docs/gmock_cook_book.md#knowing-when-to-expect-useoncall for details.
Dec 19 11:46:19.315 [INFO] RDMA device successfully deinitialized
[       OK ] RdmaTxTest.EstablishAlreadyInitialized (0 ms)
[ RUN      ] RdmaTxTest.ValidateStateTransitions
Dec 19 11:46:19.315 [DEBU] RdmaTx configure local_ip="192.168.1.10" local_port="8001" remote_ip="192.168.1.20" remote_port="8002"

GMOCK WARNING:
Uninteresting mock function call - returning default value.
    Function call: start_threads(@0x5629b645fc80 64-byte object <78-4A 4F-B4 29-56 00-00 C0-26 46-B6 29-56 00-00 A0-01 54-B4 29-56 00-00 A0-D3 45-B6 29-56 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 50-22 45-B6 29-56 00-00>)
          Returns: 4-byte object <00-00 00-00>
NOTE: You can safely ignore the above warning unless this call should not happen.  Do not suppress it by blindly adding an EXPECT_CALL() if you don't mean to enforce the call.  See https://github.com/google/googletest/blob/main/docs/gmock_cook_book.md#knowing-when-to-expect-useoncall for details.

```

Reply:
Those are just warnings, but if you prefer not to add them yet, I am fine with it

Reply:
Let Maciej add them in the next PR.